openapi.yaml

```diff
@@ -1,279 +1,720 @@
-swagger: '2.0'
+openapi: 3.0.0
 info:
-  title: Express API Skeleton
-  description: Skeleton for Express APIs
-  version: v1
+  title: Locations API
+  description: Get information about OSU locations and services
+  version: v2
   license:
     name: GNU Affero General Public License Version 3
     url: http://www.gnu.org/licenses/agpl-3.0.en.html
   contact:
     name: IS Data Architecture Team
     url: https://is.oregonstate.edu/data-architecture
     email: isdataarchitecture@oregonstate.edu
-schemes:
-  - https
-host: api.oregonstate.edu
-basePath: /v1
 externalDocs:
   description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
-produces:
-  - application/json
-consumes:
-  - application/json
+  url: https://github.com/osu-mist/locations-frontend-api
+servers:
+  - url: https://api.oregonstate.edu/v2
 security:
   - OAuth2:
       - full
 paths:
-  /pets:
+  /locations:
     get:
-      summary: List all pets
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+        - locations
+      summary: Get locations
+      description: Gets all `Location` objects. Can be filtered by parameters.
+      operationId: getLocations
       parameters:
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
         - in: query
-          name: species
-          description: Search by pet species
+          name: filter[search]
+          description: Search through buildings based on building name and abbreviation
           required: false
-          type: string
+          schema:
+            type: string
+        - in: query
+          name: filter[name]
+          description: Search buildings by name
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: filter[coordinates]
+          description: A (latitude,longitude) pair of coordinates
+          required: false
+          schema:
+            type: string
+            pattern: '^-?\d+\.\d+,-?\d+\.\d+$'
+            example: 44.567509,-123.286651
+        - in: query
+          name: filter[distance]
+          description: 'The radius of the circle centred on the specified location. Default: 2'
+          required: false
+          schema:
+            type: number
+            default: 2.0
+            format: float
+        - in: query
+          name: filter[distanceUnit]
+          description: |
+            Distance can be specified in following units:
+            - `mi` - miles
+            - `yd` - yards
+            - `ft` - feet
+            - `in` - inches
+            - `km` - kilometers
+            - `m` - meters
+            - `cm` - centimeters
+            - `mm` - millimeters
+            - `nmi` - nautical miles
+
+            Should be used with the `filter[distance]` parameter
+          required: false
+          schema:
+            type: string
+            enum: [mi, yd, ft, in, km, m, cm, mm, nmi]
```

Comment:
here's the solution for distance unit abbreviations. Looks much nicer in swagger UI right now.